hudi-client/src/main/java/org/apache/hudi/table/action/commit/CommitActionExecutor.java

```diff
@@ -115,7 +115,11 @@ public CommitActionExecutor(JavaSparkContext jsc,
   }
 
   protected HoodieMergeHandle getUpdateHandle(String partitionPath, String fileId, Iterator<HoodieRecord<T>> recordItr) {
-    return new HoodieMergeHandle<>(config, instantTime, (HoodieTable<T>)table, recordItr, partitionPath, fileId, sparkTaskContextSupplier);
+    if (table.requireSortedRecords()) {
```

Comment:
not sure if my way of thinking is right. correct me if I am wrong. we are making our file formats flexible where in either of base file or log file can be of any format. I do understand for index purposes both base file and log file will be HFile. but wondering if tagging this config to table is the right think to do or should we have two configs, one for base file and one for log file. What in case we want to experiment w/ Hfile format for log files and parquet as base files since we might get indexing capability for log files too.

Reply:
Your understanding it correct. 

I will take care of this in a separate PR with HFile implementation.